infernyx/rules.py

```diff
@@ -253,6 +281,28 @@ def tag_results(suffix, job):
             )
         }
     ),
+    InfernoRule(
+        name='activity_stream_user_counting',
+        source_tags=['incoming:activity_stream_user_counting'],
+        max_blobs=APP_MAX_BLOBS,
+        min_blobs=APP_MIN_BLOBS,
+        archive=True,
+        rule_cleanup=report_rule_stats,
+        map_input_stream=chunk_json_stream,
+        map_init_function=impression_stats_init,
+        map_keys=AS_HLL_MAP_KEYS,
+        valid_channels=['release', 'nightly', 'beta'],
+        # beta/release: 57.0, 57.0.1; nightly: 59.0a1; esr: 60.0esr, 60.0.1esr
+        version_pattern=re.compile("^\d{2}\.\w{1,5}(\.\w{1,5})?$"),
+        preprocessors=[parse_batch, assa_session_filter, partial(clean_data, imps=False), parse_ip],
+        geoip_file=GEOIP,
+        partitions=32,
+        sort_buffer_size='25%',
+        map_function=as_user_counting_map,
```

Comment:
replace the map function with:

key_parts=['date', 'release_channel', 'version', 'country_code'],
value_parts=['client_id'],